src/Attributable.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+use SilverStripe\Core\Convert;
+
+trait Attributable
+{
+    protected $attributes = [];
+
+//    private static $casting = [
+//        'AttributesHTML' => 'HTMLFragment',
+//        'getAttributesHTML' => 'HTMLFragment',
+//    ];
+
+
+
+    public function setAttribute($name, $value)
+    {
+        $this->attributes[$name] = $value;
+
+        return $this;
+    }
+
+    public function getAttribute($name)
+    {
+        $attributes = $this->getAttributes();
+
+        if (isset($attributes[$name])) {
+            return $attributes[$name];
+        }
+
+        return null;
+    }
+
+    protected abstract function getDefaultAttributes(): array;
+
+    /**
+     * Allows customization through an 'updateAttributes' hook on the base class.
+     * Existing attributes are passed in as the first argument and can be manipulated,
+     * but any attributes added through a subclass implementation won't be included.
+     *
+     * @return array
+     */
+    public function getAttributes()
```

Comment:
Is, the intention here to replace things like this? https://github.com/silverstripe/silverstripe-framework/blob/4/src/Forms/CheckboxField.php#L29

Reply:
My initial intention would be to replace only the `FormField` one. All the other other ones are just overriding the FormField one.

I think there's a couple other ViewableData that reimplements this Attribute pattern. 

Possible next step would be:
1. Delete this trait and copy all the logic straight into ImageManipulation.
2. Move this trait and move it to framework. Initally only the ImageManipulation trait would reference it.
3. Step 2 + update FormFiled to use it, but nothing else.

Don't have any preference at this stage. Just want to get this over the line in time.

Reply:
At this stage I'd say option 1. since that fixes the immediate need